cli/azd/pkg/azsdk/client_options_builder.go

```diff
@@ -0,0 +1,52 @@
+package azsdk
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type ClientOptionsBuilder struct {
+	transport policy.Transporter
+	policies  []policy.Policy
+}
+
+func NewClientOptionsBuilder() *ClientOptionsBuilder {
+	return &ClientOptionsBuilder{}
+}
+
+// Sets the underlying transport used for executing HTTP requests
+func (b *ClientOptionsBuilder) WithTransport(transport policy.Transporter) *ClientOptionsBuilder {
+	b.transport = transport
+	return b
+}
+
+// Appends policies into the HTTP pipeline
+func (b *ClientOptionsBuilder) WithPolicy(policy policy.Policy) *ClientOptionsBuilder {
+	b.policies = append(b.policies, policy)
+	return b
+}
```

Comment:
policies can be set by call or by retry.  
We need to either:
- have two methods
- have an argument to define where the policy is added